### PR TITLE
A new class, Renderer, to gather code related to rendering and coordinate transforms

### PR DESCRIPTION
**Pull request by pleroy**
This is the first chunk of the ill-fated #1374.  Renderer is (moderately) tested but not used in production code yet.

### CODE REVIEW

File: serialization/ksp_plugin.proto [hunk 1]
```diff
@@ -111,6 +111,11 @@ message Plugin {
   reserved "/*celestial*/";
 }
 
+// Added in Catalan.
```
Comment (eggrobin) [inline serialization/ksp_plugin.proto]: Cauchy.

File: ksp_plugin/renderer.cpp [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+bool Renderer::HasTargetVessel() const {
+  return static_cast<bool>(target_);
+}
+
+Vessel& Renderer::GetTargetVessel() {
+  CHECK(target_);
+  return *target_->vessel;
+}
+
+Vessel const & Renderer::GetTargetVessel() const {
```
Comment (eggrobin) [inline ksp_plugin/renderer.cpp]: ...

File: serialization/ksp_plugin.proto (hunk 1 above)
Comment (pleroy) [inline serialization/ksp_plugin.proto]: Done.

File: ksp_plugin/renderer.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/renderer.cpp]: Done.

**Comment by pleroy** [comment]
Retest this please

**Pull request merged**